HuggingFace & Cloudflare WebRTC partnership announcement (#2801)
huggingface/blog: fastrtc-cloudflare.md

---
title: "Hugging Face and Cloudflare Partner to Make Real-Time Speech and Video Seamless with FastRTC" 
thumbnail: /blog/assets/fastrtc-cloudflare/fastrtc_cloudflare.png
authors:
- user: freddyaboulton
---

# Hugging Face and Cloudflare Partner to Make Real-Time Speech and Video Seamless with FastRTC

We're excited to announce a new partnership between Cloudflare and Hugging Face that gives FastRTC developers instant access to enterprise-grade WebRTC infrastructure with a Hugging Face token.

> As a preview of what you can build with FastRTC and Cloudflare, check out this voice chat app built with Meta's new Llama 4 model!

Reviewer: nice to build w/ llama 4

> <video src="https://huggingface.co/datasets/freddyaboulton/bucket/resolve/main/FastRTCCloudFlareShort.mp4" controls /></video>

Reviewer: btw the video did not load for me, but my wifi connection is pretty terrible atm. but maybe double check the link

Author: Thanks for catching this. My first draft had a video loading from an HF dataset and that was working but for some reason this particular video is not loading. Switched to loading from a github url and it's working now.

## Free Access with Your Hugging Face Account

FastRTC developers with a valid Hugging Face Access Token can stream 10GB of data for FREE every month without a credit card. Once the monthly limit is reached, developers can switch to their Cloudflare account for higher capacity ([instructions](https://fastrtc.org/deployment/#cloudflare-calls-api)). 

## Why This Matters for AI Developers

This partnership is especially valuable for AI developers building:
- Voice assistants that need reliable, low-latency audio streaming
- Video analysis applications that process camera feeds in real-time
- Multimodal AI applications that combine audio, video, and text

This partnership lets developers focus on their core application logic with FastRTC, while eliminating the need to build and maintain TURN infrastructure. Cloudflare's managed service handles global scalability and reliability, allowing AI developers to deliver exceptional experiences without the overhead of maintaining infrastructure.

## Getting Started

> The integration will be available in the FastRTC version 0.0.20. To get started:

Reviewer: will the release be out by the time the blog post is released? if so I'd drop the future tense and instead say:
```suggestion
The integration is available with `fastrtc>=0.0.20`. To get started:
```

> - Ensure you have a Hugging Face token (get one [here](https://huggingface.co/settings/tokens))

Reviewer: ```suggestion
- Ensure you have a Hugging Face token with read access (get one [here](https://huggingface.co/settings/tokens))
```

- Install or upgrade FastRTC: `pip install --upgrade 'fastrtc[vad]'`

> - Configure your Stream to use the Cloudflare TURN network
> - Launch your script with python

Reviewer: ```suggestion
- Configure your Stream to use the Cloudflare TURN network as shown below:
```

```python
from fastrtc import ReplyOnPause, Stream, get_cloudflare_turn_credentials
import os

os.environ["HF_TOKEN"] = "<your-hf-token>"

def echo(audio):
    yield audio

stream = Stream(ReplyOnPause(echo),
                rtc_config=get_cloudflare_turn_credentials)
stream.ui.launch()
```

> See this [Collection](https://huggingface.co/collections/fastrtc/cloudflare-partnership-67f437e0dfd19818d62ccb81) on Hugging Face as well as the [FastRTC Cookbook](https://fastrtc.org/cookbook/) for more examples.

Reviewer: ```suggestion

Now simply launch your script with Python: `python [name of your file]`

See this [Collection](https://huggingface.co/collections/fastrtc/cloudflare-partnership-67f437e0dfd19818d62ccb81) on Hugging Face as well as the [FastRTC Cookbook](https://fastrtc.org/cookbook/) for more examples.
```
